src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/VirtualProject.cs

```diff
@@ -21,6 +24,46 @@ internal static class VirtualCSharpFileBasedProgramProject
     internal static string GetVirtualProjectPath(string documentFilePath)
         => Path.ChangeExtension(documentFilePath, ".csproj");
 
+#region TODO: Copy-pasted from dotnet run-api. Delete when run-api is adopted.
+    private static class Sha256Hasher
```

Comment:
Need to have an issue tracking this as all of our crypto use needs to be centralized (when possible). We should move this to one of those existing mechanisms when we have a bit more time.

Reply:
Filed #78618 to track deleting this code.

Surprised I did not see an issue for adopting run-api already.